src/porepy/numerics/mixed_dim/coupler.py

```diff
@@ -101,13 +112,24 @@ def split(self, gb, key, values):
 
         """
         dofs = np.empty(gb.size(), dtype=int)
-        for _, d in gb:
-            dofs[d['node_number']] = d['dof']
+        for _, data in gb:
+            k_data = self.__get_data(data, data_key)
+            dofs[data['node_number']] = k_data['dof']
         dofs = np.r_[0, np.cumsum(dofs)]
 
-        gb.add_node_prop(key)
+        if data_key is None:
+            gb.add_node_prop(key)
+
         for g, d in gb:
-            i = d['node_number']
-            d[key] = values[slice(dofs[i], dofs[i+1])]
+            k_data = self.__get_data(data, data_key)
+            i = data['node_number']
+            k_data[key] = values[slice(dofs[i], dofs[i+1])]
 
 #------------------------------------------------------------------------------#
+    def __get_data(self, data, data_key):
```

Comment:
a short example of how this would be used: 
` gb = create_grid()
  # Assign parameters
    gb.add_node_props(['flow', 'elasticity'])
    for g, d in gb:
        if g.dim==3:
            elasticity_dict = _get_data_elasticity(g)
            d['elasticity'] = elasticity_dict
        flow_dict = _get_data_flow(g)
        d['flow'] = flow_dict
   # Choose and define the solvers and coupler
    solver_flow = mpfa.Mpfa()
    solver_elasticity = fvf.FracturedMpsa()

    coupling_conditions = tpfa_coupling.TpfaCoupling(solver_flow)
    solver_flow = coupler.Coupler(solver_flow, coupling_conditions)
    A, b = solver_flow.matrix_rhs(gb, data_key='flow')
`

So, the user is responsible to define and give the correct key-word. We could fix the subdict names for different solvers (e.g., flow, transport, mech), but this would loose a bit of the generality. What if I want to run two flow simulations on the same bucket, but with different fracture apertures? 

As you say, if we don't need subdict we can just drop it and store the data directly in the `data`